python/cuspatial/cuspatial/core/binpreds/feature_covers.py

```diff
@@ -36,21 +49,58 @@ class CoversPredicateBase(EqualsPredicateBase):
     pass
 
 
+class LineStringLineStringCovers(BinPred):
+    def _preprocess(self, lhs, rhs):
+        return _basic_equals_all(rhs, lhs)
+
+
+class PolygonPointCovers(BinPred):
+    def _preprocess(self, lhs, rhs):
+        return _basic_contains_any(lhs, rhs)
+
+
+class PolygonLineStringCovers(BinPred):
+    def _preprocess(self, lhs, rhs):
```

Comment:
Would be nice to have some docs.

Reply:
Since your fix for degenerate results is in, I think this section is over engineered. I have removed the conditional case that handled degenerate linestring results and added a couple of comments that improve comprehension.